src/Mod/Sketcher/Gui/Command.cpp

```diff
@@ -1271,32 +1271,26 @@ class SnapSpaceAction : public QWidgetAction
         languageChange();
 
         QObject::connect(snapToObjects, &QCheckBox::stateChanged, [this](int state) {
-            ParameterGrp::handle hGrp = App::GetApplication().GetParameterGroupByPath("User parameter:BaseApp/Preferences/Mod/Sketcher/Snap");
+            ParameterGrp::handle hGrp = this->getParameterPath();
             hGrp->SetBool("SnapToObjects", state == Qt::Checked);
         });
 
         QObject::connect(snapToGrid, &QCheckBox::stateChanged, [this](int state) {
-            ParameterGrp::handle hGrp = App::GetApplication().GetParameterGroupByPath("User parameter:BaseApp/Preferences/Mod/Sketcher/Snap");
+            ParameterGrp::handle hGrp = this->getParameterPath();
             hGrp->SetBool("SnapToGrid", state == Qt::Checked);
         });
 
         QObject::connect(snapAngle, qOverload<double>(&Gui::QuantitySpinBox::valueChanged), [this](double val) {
-            ParameterGrp::handle hGrp = App::GetApplication().GetParameterGroupByPath("User parameter:BaseApp/Preferences/Mod/Sketcher/Snap");
+            ParameterGrp::handle hGrp = this->getParameterPath();
             hGrp->SetFloat("SnapAngle", val);
         });
 
         return snapW;
     }
 
 private:
-    ViewProviderSketch* getView() {
-        Gui::Document* doc = Gui::Application::Instance->activeDocument();
-
-        if (doc) {
-            return dynamic_cast<SketcherGui::ViewProviderSketch*>(doc->getInEdit());
-        }
-
-        return nullptr;
+    ParameterGrp::handle getParameterPath() {

```

Comment:
Should we here use static as we did the last time ?

Reply:
This is a very good question.

I do not think it is worth it. 

It makes a lot of sense to encapsulate it in a single function, so that if the group changes, there is a single place. Then this is a very short function and is defined on the class declaration which means it is implicitly inline. The compiler will almost surely expand it and optimise. Then, the overhead is only the one of the function inside. In cases where in addition we are observing the group, the usage is sporadic.

By making it static, the value itself is stored and you remove that overhead. However, a static variable is allocated when the program starts (save for local static which depends on the first call to the function where it is local), and it will be allocated until the program ends, even if no object is allocated of that type (when it is a member function). This means that memory cannot be used elsewhere, even if idle.

So when making something static it makes sense to think what is the benefit. Sometimes it may be a huge performance improvement, others simplicity in management of resources, others you need an inherently thread-safe type (which you can achieve with local storage to a function, as in the Meyer's singleton). I do not have a table mapping when yes and when no. I think we should decide on a case-by-case basis.

Reply:
Thanks